Topics/Tech_Stacks/selenium.md

```diff
@@ -124,6 +124,149 @@ driver.quit()
+    from selenium.webdriver.common.by import By
+    from selenium.webdriver.support.ui import WebDriverWait
+    from selenium.webdriver.support import expected_conditions as EC
+    import time
+
+    # The Google Form we are working with
+    browserURL = "https://forms.gle/rbxs4fNCbBSzfPW1A"
+
+    # Make sure the driver version matches your Chrome browser
+    # Can be found at https://googlechromelabs.github.io/chrome-for-testing/
+    pathToChromeDriver = "C:\\Users\\edwar\\Downloads\\chromedriver.exe"
+
+
+    def fill_form(browser):
+        pass
+
+
+    if __name__ == '__main__':
+    # Set up ChromeDriver service
+    options = Service(executable_path=pathToChromeDriver)
+
+    # Initialize WebDriver with ChromeDriver service
+    driver = webdriver.Chrome(service=options)
+    driver.get(browserURL)
+
+    # Call function to populate the form
+    fill_form(driver)
+
+    # Stop the webdriver after form has been submitted
+    driver.quit()
+
+    ```
+    The function ```fill_form``` is what we will be focused on working on. If you were to execute the current file, you would see Chrome opening up, it navigating to the Google form followed by it closing.
+
+2.  **Selenium: Populating Text Fields**
+
+    To accomplish this, we will need to do the following - **(a) locate the element which we want to populate, (b) wait for it to be visible, (c) populate the field**. This is the general structure to be followed when doing UI Automation and we will follow this for all the fields we will populate.
+
+    To locate the element we will use css selectors which is essentially the same as how we style css elements by locating them on css files. Inspecting the Text Field element we see the following:
+
+    ![alt text](SeleniumImages/textField.png)
+
+    We see that the text field element has ```class = "whsOnd zHQkBf"``` so we will use this as the css selector for this element.
+
+    After locating the css selector we will need to wait for that element to be visible and then we need to populate it with some text. We will do this in our ```fill_form``` function and below is an example of how we can achieve this.
+
+    ```python
+    # Text Field
+    textField = WebDriverWait(driver, 5).until(
+        EC.visibility_of_element_located((By.CLASS_NAME, "whsOnd.zHQkBf"))
+    )
+    textField.click()
+    textField.send_keys("Hello")
+    ```
+3.  **Selenium: Selecting Multiple Choice**
+    
+    Let's say we wanted to select 'Option 2'. Following similar steps as above we will first locate the css selectors. Inspecting the multiple choice option for 'Option 2' we see the following:
+
+    ![alt text](SeleniumImages/multipleChoice.png)
+
+    We see there is a unique id field ```id = "i12"```which we can use for our css selector. We want to ensure that our css selectors are unique so that we choose only the item we want to select. As such below is the example code as how we could go about doing this.
```

Comment:
I would suggest to add a bit more information on how to identify/choose what field to include in the CSS selector, as some of the fields can be dynamically generated as part of the JavaScript code.  I.e., The third option under a <div> with classname="something"